crates/bevy_color/src/color.rs

```diff
@@ -37,6 +43,49 @@ impl Default for Color {
     }
 }
 
+impl Alpha for Color {
+    fn with_alpha(&self, alpha: f32) -> Self {
+        let mut new = *self;
+
+        match &mut new {
+            Color::Srgba(x) => *x = x.with_alpha(alpha),
+            Color::LinearRgba(x) => *x = x.with_alpha(alpha),
+            Color::Hsla(x) => *x = x.with_alpha(alpha),
+            Color::Lcha(x) => *x = x.with_alpha(alpha),
+            Color::Oklaba(x) => *x = x.with_alpha(alpha),
+        }
+
+        new
+    }
+
+    fn alpha(&self) -> f32 {
+        match self {
+            Color::Srgba(x) => x.alpha(),
+            Color::LinearRgba(x) => x.alpha(),
+            Color::Hsla(x) => x.alpha(),
+            Color::Lcha(x) => x.alpha(),
+            Color::Oklaba(x) => x.alpha(),
+        }
+    }
+}
+
+impl Mix for Color {
+    fn mix(&self, other: &Self, factor: f32) -> Self {
```

Comment:
@alice-i-cecile and I discussed this, I think we would like to avoid magical implicit conversions. The problem is that they can be either fast or slow, depending on the data, and this may be surprising if the user is not careful.

Reply:
Fair enough! I've removed this implementation and also removed it as a requirement for `StandardColor` as well.